### PR TITLE
OSDOCS-1086: Adding docs for graceful shutdown and restart

### PR DESCRIPTION
**Pull request by bergerhoffer**
https://issues.redhat.com/browse/OSDOCS-1086

Preview: https://osdocs-1086-shutdown--ocpdocs.netlify.app/openshift-enterprise/latest/backup_and_restore/graceful-cluster-shutdown.html

### CODE REVIEW

**Comment by bergerhoffer** [comment]
@xingxingxia @rpattath I believe that this is ready for QE review whenever you are ready. Though I believe there is still some testing of a 30 day shutdown that needs to be done, it would still be good to have this ready for whenever that is complete. Let me know if you have any questions for feedback, thanks!

Preview:
* https://osdocs-1086-shutdown--ocpdocs.netlify.app/openshift-enterprise/latest/backup_and_restore/graceful-cluster-shutdown.html
* https://osdocs-1086-shutdown--ocpdocs.netlify.app/openshift-enterprise/latest/backup_and_restore/graceful-cluster-restart.html

File: modules/graceful-restart.adoc [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+The worker nodes are ready if the status is `Ready`, as shown in the following output:
++
+----
+NAME                           STATUS   ROLES    AGE   VERSION
+ip-10-0-179-95.ec2.internal    Ready    worker   64m   v1.18.3
+ip-10-0-182-134.ec2.internal   Ready    worker   64m   v1.18.3
+ip-10-0-250-100.ec2.internal   Ready    worker   64m   v1.18.3
+----
+
+. If the worker nodes are _not_ ready, then check whether there are any pending certificate signing requests that must be approved.
```
Comment (xingxingxia) [inline modules/graceful-restart.adoc:55]: > If the worker nodes
This sentence seems not only applies worker nodes, applies to master nodes as well.

**Comment by xingxingxia** [comment]
Looks good from my side with a minor comment inline.

**Comment by rpattath** [comment]
LGTM

File: modules/graceful-restart.adoc (hunk 1 above)
Comment (bergerhoffer) [reply modules/graceful-restart.adoc:55]: Good point. @vareti @sttts Is there a chance that the master nodes will need to have CSRs approved before they become ready? Like, that we need to add the steps to check for and approve CSRs for the masters to step 3?

https://osdocs-1086-shutdown--ocpdocs.netlify.app/openshift-enterprise/latest/backup_and_restore/graceful-cluster-restart.html#graceful-restart_graceful-restart-cluster

Comment (vareti) [reply modules/graceful-restart.adoc:55]: @xingxingxia is right. I just came to know that it is possible. We could have similar steps to check CSR for master nodes too.

File: modules/graceful-shutdown.adoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+done
+----
++
+Shutting down the nodes using one of these methods allows pods to terminate gracefully, which reduces the chance for data corruption.
++
+[NOTE]
+====
+It is not necessary to drain master nodes for standard workloads prior to shutdown. If you do, you must mark the master nodes as schedulable before the cluster will be functional again after restart.
+
+Cluster administrators are responsible for ensuring a clean restart of their workloads after the cluster is restarted.
```
Comment (sttts) [inline modules/graceful-shutdown.adoc:39]: their own

**Comment by bergerhoffer** [comment]
@xingxingxia @sttts @vareti I've updated to include the CSR approval steps for master nodes, and to add a note on draining (requested by https://bugzilla.redhat.com/show_bug.cgi?id=1802836#c12).

Can you all please take another look?

* https://osdocs-1086-shutdown--ocpdocs.netlify.app/openshift-enterprise/latest/backup_and_restore/graceful-cluster-restart.html
* https://osdocs-1086-shutdown--ocpdocs.netlify.app/openshift-enterprise/latest/backup_and_restore/graceful-cluster-shutdown.html#graceful-shutdown_graceful-shutdown-cluster

**Comment by vareti** [comment]
looks good.

**Review by lamek — COMMENTED** [review]
This looks great.

One thought - should we include a note or a line about what to do in the event that you cannot connect with `oc` after restarting your nodes?
https://github.com/openshift/openshift-docs/pull/23028/files#diff-9a3b0516c75db5c1a3fd397b561723eaR25

**Comment by bergerhoffer** [comment]
> This looks great.
> 
> One thought - should we include a note or a line about what to do in the event that you cannot connect with `oc` after restarting your nodes?
> https://github.com/openshift/openshift-docs/pull/23028/files#diff-9a3b0516c75db5c1a3fd397b561723eaR25

@vareti @sttts Is this something that a user might run into that we should provide guidance for?

**Comment by vareti** [comment]
> Is this something that a user might run into that we should provide guidance for?

@bergerhoffer isn't that what we are saying [here](https://github.com/openshift/openshift-docs/pull/23028/files#diff-6d68b7460bea0c71d3d048beab596ce5R12-R14)?

**Comment by bergerhoffer** [comment]
> > Is this something that a user might run into that we should provide guidance for?
> 
> @bergerhoffer isn't that what we are saying [here](https://github.com/openshift/openshift-docs/pull/23028/files#diff-6d68b7460bea0c71d3d048beab596ce5R12-R14)?

Yes that's true, I forgot about that. That should be sufficient then, thanks!

**Comment by bergerhoffer** [comment]
Hi @xingxingxia @rpattath, can you let me know where we are with finalizing testing for this? I believe we were waiting on testing of a 30 day shutdown, but I wasn't sure how close we were to this. Thanks!

**Comment by xingxingxia** [comment]
I already replied in https://issues.redhat.com/browse/OSDOCS-1086 comments

**Comment by bergerhoffer** [comment]
Hi @rpattath - are there any updates on this?

**Comment by bergerhoffer** [comment]
QE approval received in Jira, merging!

**Pull request merged**